src/libraries/System.Linq/src/System/Linq/OrderBy.cs

```diff
@@ -132,6 +137,17 @@ public static partial class Enumerable
 
             return source.CreateOrderedEnumerable(keySelector, comparer, true);
         }
+
+        /// <summary>Gets whether the results of an unstable sort will be observably the same as a stable sort.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool TypeIsImplicitlyStable<T>() =>
+            typeof(T) == typeof(sbyte) || typeof(T) == typeof(byte) ||
+            typeof(T) == typeof(int) || typeof(T) == typeof(uint) ||
+            typeof(T) == typeof(short) || typeof(T) == typeof(ushort) ||
+            typeof(T) == typeof(long) || typeof(T) == typeof(ulong) ||
+            typeof(T) == typeof(Int128) || typeof(T) == typeof(UInt128) ||
+            typeof(T) == typeof(nint) || typeof(T) == typeof(nuint) ||
+            typeof(T) == typeof(bool) || typeof(T) == typeof(char);
```

Comment:
blink twice if you want us to intrinsify `typeof().IsPrimitive` 😄

Reply:
Heh.  It would still end up being something like:
```C#
(typeof(T).IsPrimitive && typeof(T) != typeof(float) && typeof(T) != typeof(double)) ||
typeof(T) == typeof(Int128) || typeof(T) == typeof(UInt128)
```